File: apis/src/main/java/com/beat/apis/performance/application/dto/modify/staff/StaffModifyResponse.java
package com.beat.apis.performance.application.dto.modify.staff;

import com.beat.global.support.jackson.CdnImageUrl;

public record StaffModifyResponse(
	Long staffId,
	String staffName,
	String staffRole,
	@CdnImageUrl String staffPhoto) {
	public static StaffModifyResponse of(Long staffId, String staffName, String staffRole, String staffPhoto) {
		return new StaffModifyResponse(staffId, staffName, staffRole, staffPhoto);
	}
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/StaffModifyResponse.kt
package com.beat.apis.performance.api.response

import com.beat.apis.performance.application.result.StaffResult
import com.beat.global.support.jackson.CdnImageUrl

@ConsistentCopyVisibility
data class StaffModifyResponse private constructor(
    val staffId: Long?,
    val staffName: String?,
    val staffRole: String?,
    @field:CdnImageUrl val staffPhoto: String?,
) {
    companion object {
        fun from(result: StaffResult): StaffModifyResponse = StaffModifyResponse(result.id, result.name, result.role, result.photo)
    }
}
